tests/st/test_pool.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2015-2016 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from test_base import TestBase
+from tests.st.utils.docker_host import DockerHost
+from tests.st.utils.exceptions import CommandExecError
+
+"""
+Test calicoctl pool
+
+1) Test the CRUD aspects of the pool commands.
+2) Test IP assignment from pool.
+
+BGP exported routes are hard to test and aren't expected to change much so
+write tests for them (yet)
+
+"""
+
+
+class TestPool(TestBase):
+    def test_pool_crud(self):
+        """
+        Test that a basic CRUD flow for pool commands.
+        """
+        with DockerHost('host', dind=False, start_calico=False) as host:
+            # Set up the ipv4 and ipv6 pools to use
+            ipv4_pool = "10.0.1.0/24"
+            ipv6_pool = "fed0:8001::/64"
+
+            ipv4_yaml = """
```

Comment:
We've already got pyyaml as a requirement - makes sense to use dictionaries, but write out using pyyaml.